PowerShellGet/PSModule.psm1

```diff
@@ -11633,26 +12185,98 @@ function Get-InstalledModuleDetails
-                                                                if($RequiredVersion)
-                                                                {
-                                                                   if($RequiredVersion -eq $InstalledModuleDetails.PSGetItemInfo.Version)
-                                                                   {
-                                                                       $InstalledModuleDetails
-                                                                   }
-                                                                }
-                                                                else
-                                                                {
-                                                                    if( (-not $MinimumVersion -or ($MinimumVersion -le $InstalledModuleDetails.PSGetItemInfo.Version)) -and 
-                                                                        (-not $MaximumVersion -or ($MaximumVersion -ge $InstalledModuleDetails.PSGetItemInfo.Version)))
-                                                                    {
-                                                                        $InstalledModuleDetails
-                                                                    }
-                                                                }
+
+                                                                Test-ItemPrereleaseVersionRequirements -InstalledItemDetails $InstalledModuleDetails `
+                                                                                                            -RequiredVersion $RequiredVersion `
+                                                                                                            -MinimumVersion $MinimumVersion `
+                                                                                                            -MaximumVersion $MaximumVersion
                                                             }
                                                         }
                                                     }
 }
 
+# Returns the installed item details object if it meets the Required, Minimum, and Maximum version bounds.
+function Test-ItemPrereleaseVersionRequirements
+{
+    [CmdletBinding()]
+    param(
+
+        [ValidateNotNullOrEmpty()]
+        [System.Collections.Hashtable]
+        $InstalledItemDetails,
+
+        [string]
+        $RequiredVersion,
+        
+        [string]
+        $MinimumVersion,
+
+        [string]
+        $MaximumVersion
+    )
+
+    $psgitemVersionPart,$psgitemPrereleasePart = $InstalledItemDetails.PSGetItemInfo.Version -split '-',2
+
+    if($RequiredVersion)
+    {
+        $requiredVersionPart,$requiredPrereleasePart = $RequiredVersion -split '-',2
+        
+        if (($requiredVersionPart -eq $psgitemVersionPart) -and 
+            ($requiredPrereleasePart -eq $psgitemPrereleasePart)) 

```

Comment:
>$requiredPrereleasePart -eq $psgitemPrereleasePart [](start = 13, length = 50)

This may not work as expected when $requiredversion with prerelease string ends with whitespace.